### PR TITLE
Api controller and routes for tags, comments, cards

### PR DESCRIPTION
**Pull request by tsaputo**
Added api controller and routes for:

- tags
- comments
- cards

### CODE REVIEW

File: app/controllers/api/v1/cards_controller.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      before_action :find_column!, except: :index
+      before_action :find_card!, except: %i[index new create]
+      before_action :find_board!, only: [:update]
+      helper_method :sort_column, :sort_direction, :sort_filter
+      before_action :find_user_by_email!, only: %i[add_assignee]
+
+      #FIXME:as soon as current user is available
+      def index
+        @load_new = params[:load_new] || false
+        @cards = Card.all
```
Comment (yuravenzhik) [inline app/controllers/api/v1/cards_controller.rb:13]: pagination

File: app/controllers/api/v1/cards_controller.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        end
+      end
+
+      # { "email": "me@gmail.com" }
+      def add_assignee
+        membership = @column.board.memberships.find_by(user: @user)
+        unless membership
+          render json: { error: 'User is not a member of this board' }, status: 404
+          return
+        end
```
Comment (yuravenzhik) [inline app/controllers/api/v1/cards_controller.rb:66]: before_action

File: app/controllers/api/v1/cards_controller.rb [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        else
+          @error_message = @card.errors.full_messages
+          render :error_message, status: 422
+        end
+      end
+
+      # "user_id": 2
+      def create
+        @card = @column.cards.build(card_params)
+        if @card.save
```
Comment (yuravenzhik) [inline app/controllers/api/v1/cards_controller.rb:41]: !

File: app/controllers/api/v1/cards_controller.rb [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+          render :error_message, status: 422
+        end
+      end
+
+      # { "target_cards_id": "[8,4,9,10]","source_cards_id": "[5,6,7]" }
+      # { "target_cards_id": "[8,4,9,10]","source_cards_id": "[]" }
+      def update_position
+        result, errors = Cards::UpdatePositionService.call(@card, @column, params[:target_cards_id], params[:source_cards_id])
+        if result
+          @message = 'Card position was successfully updated'
```
Comment (yuravenzhik) [inline app/controllers/api/v1/cards_controller.rb:63]: render :show

File: app/controllers/api/v1/cards_controller.rb [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+          render :success_message, status: 200
+        else
+          @error_message = errors.full_messages
+          render :error_message, status: 422
+        end
+      end
+
+      def destroy
+        if @card.discard
+          @message = 'Card was successfully discarded!'
```
Comment (yuravenzhik) [inline app/controllers/api/v1/cards_controller.rb:73]: deleted

File: app/controllers/api/v1/comments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  module V1
+    class CommentsController < Api::V1::BaseController
+      load_and_authorize_resource :comment, except: :create
+      before_action :find_column!
+      before_action :find_comment!, only: [:show, :update, :destroy]
+      before_action :find_card!
+
+      def index
+        @comments = @card.comments.order(created_at: :desc).paginate(page: params[:page])
+        render :index, status: 200
```
Comment (yuravenzhik) [inline app/controllers/api/v1/comments_controller.rb:11]: -

File: app/controllers/api/v1/comments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      end
+
+      def show
+        render :show, status: 200
```
Comment (yuravenzhik) [inline app/controllers/api/v1/comments_controller.rb:15]: -

File: app/controllers/api/v1/comments_controller.rb [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      end
+
+      def create
+        @comment = @card.comments.build(comment_params)
+        if @comment.save
```
Comment (yuravenzhik) [inline app/controllers/api/v1/comments_controller.rb:20]: create

File: app/controllers/api/v1/tags_controller.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+      before_action :find_card!
+      before_action :find_tag!, only: [:destroy, :show]
+
+      def index
+        @tags = @card.tags
+        render :index, status: 200
+      end
+
+      def show
+        render :show, status: 200
```
Comment (yuravenzhik) [inline app/controllers/api/v1/tags_controller.rb:14]: !

File: app/views/api/v1/cards/_author.json.jbuilder [hunk 10]
```diff
@@ -0,0 +1,2 @@
+json.full_name user.full_name
+json.email user.email
```
Comment (yuravenzhik) [inline app/views/api/v1/cards/_author.json.jbuilder:2]: json.(user, :first_name, :last_name, :email, :avatar_url)

File: app/views/api/v1/cards/edit.json.jbuilder [hunk 11]
```diff
@@ -0,0 +1,4 @@
+json.title @card.title
```
Comment (yuravenzhik) [inline app/views/api/v1/cards/edit.json.jbuilder:1]: !

File: app/views/api/v1/cards/card.json.jbuilder [hunk 12]
```diff
@@ -0,0 +1,9 @@
+json.title @card.title
```
Comment (yuravenzhik) [inline app/views/api/v1/cards/card.json.jbuilder:1]: !

File: app/views/api/v1/cards/index.json.jbuilder [hunk 13]
```diff
@@ -0,0 +1,12 @@
+json.array!(@cards) do |card|
+  json.title card.title
```
Comment (yuravenzhik) [inline app/views/api/v1/cards/index.json.jbuilder:2]: в паршиал и использовать с show

File: app/views/api/v1/comments/index.json.jbuilder [hunk 14]
```diff
@@ -0,0 +1,4 @@
+json.array!(@comments) do |comment|
+  json.body comment.body
```
Comment (yuravenzhik) [inline app/views/api/v1/comments/index.json.jbuilder:2]: + created_at, updated_at, id и в паршиал

File: spec/requests/api/v1/cards_request_spec.rb [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+      end
+      it "renders correct template" do
+        subject
+        expect(subject).to render_template(:show)
+      end
+
+      it "returns json with correct title" do
+        subject
+        expect(JSON.parse(response.body)).to include_json(
+          title: "696"
```
Comment (yuravenzhik) [inline spec/requests/api/v1/cards_request_spec.rb:130]: let

File: app/controllers/api/v1/cards_controller.rb [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+                .filter_by_board(params[:board_title])
+        paginate_cards
+      end
+
+      def show; end
+
+      def create
+        @card = @column.cards.build(card_params)
+        if @card.save
+          render :show, status: 200
```
Comment (etherell) [inline app/controllers/api/v1/cards_controller.rb:32]: Status 201?

**Pull request merged**